### ISSUE TITLE
conducting three-way ANOVA unequal group sizes

### ISSUE THREAD

**Issue by MKeller96** [0todd0000/spm1d#159]
Hi todd,

I would like to conduct a three-way ANOVA (6x2x2) with one RM factor using; spm1d.stats.anova3onerm(Y, A, B, C, SUBJ). Sadly I get an error saying ''Design must be balanced.'' The SPM1d documentation (Matlab) says that unbalanced two- and three-way repeated-measures ANOVA results have not been verified and thus results have to be taken with caution. Therefore I thought it was possible to run this function for unequal group sizes. However, in another post I read that this is only supported for one way ANOVA's. I would still like to do this, am I doing something wrong?

Y = data set
A = muscle          
(I am analysing 6 muscles' RMS amplitude per subject)
B = group             
(I have 2 groups, one with 9 subjects and one with 22 subjects)
C = time               
(I measured the RMS amplitude at two points in time)

I already ran a 2-way RM ANOVA (6x2) for both groups seperatly which worked fine. I hope you can help me! 

Kind regards,

Max Keller

**Comment by 0todd0000** [comment]
Hi Max,

Unbalanced designs are indeed not yet supported for general ANOVA in **spm1d**. The necessary non-sphericity corrections have been implemented, but they are not yet working correctly for continuum-level results.

Running the two-way RM analyses worked likely because the design is balanced when considering one group in isolation. 

One possibility to proceed would be to:
1. Randomly select 9 subjects from the second group.
2. Run balanced ANOVA.
3. Repeat 1 and 2 for a number of random sub-samples (at least 20, and the more the better);  if the results are qualitatively stable over random sub-samples  it would suggest that the larger sample is not necessarily needed to achieve the same results.

This is a non-ideal work-around, but the results would be robust if stable to random sub-sampling.

Todd

**Comment by JoepSus** [comment]
Hi Todd and Mac,

I would like to shear my experiences as I encountered the same problem using ANOVA with unequally balanced samples and receiving the error ‘Design must be balanced’.  

Summary: An EMG study, Intervention group (n=11) vs Non-intervention group (n=13), measurement Pre and Post (n=2), three muscles (n=3), 0-100% extent of a movement. 

My solution: For bootstrapping I created a ‘bootstrapmatrix’ of 11 x n_samples_bootstrap with uniform distribution function 'randi()', randomly filled with subject numbers of the intervention group: ‘randBOOT = randi([12 24],11,10000);’. Then, a for loop created per loop balanced data, following the subjects of the intervention group from the ‘bootstrapmatrix. Using:

spm1d.stats.anova3tworm(anova.Y, anova.A, anova.B, anova.C, anova.S)

Y = Data (132x100)
A = Intervention 	1x(2x66)
B = Visit 		        2x(2x33)
C = Muscle 		3x(3x11)
S = Subject_log	6x(1:22)

At the end of each loop, I exported the main effects for that bootstrap sample (spmilist.SPMs{1, 1}.h0reject ). Now I know in how many cases of the 10000 or 1000 samples the percentage of main effects that were found. The output was roughly the same.

Kind regards,
Joep :)
